agents-audit/src/main/java/org/apache/ranger/audit/utils/AbstractRangerAuditWriter.java

```diff
@@ -262,10 +263,14 @@ public PrintWriter createWriter() throws Exception {
         }
 
         if (logWriter == null) {
-            // Create the file to write
-            logger.info("Creating new log file. auditPath=" + fullPath);
-            createFileSystemFolders();
-            ostream               = fileSystem.create(auditPath);
+            if (reUseLastLogFile){
+                ostream = fileSystem.append(auditPath);
```

Comment:
In case of failure in writing to audit file, the last audit log record may be incomplete. If subsequent records are appended to this file, all subsequent records may not be processed by the reader - hence the decision to use a new file. It will be safe to append only when the last record in the file is a complete JSON.

Reply:
@mneethiraj 
I observed this while testing the patch: In case of write failures, flush() is never called (as stream becomes null) ensuring writes to audit log are atomic in nature(always complete json). The switch happens to spool file for the writes which don't go through and then when the destination is up again content from spool is written back to it.

Reply:
@kumaab What about if the writing to the spool directory itself gets failed(assume some storage issue)

Reply:
@bhavikpatel9977 Spool is used as a backup when destination is not reachable, if spool is not able to store it is expected to fail.